Chime/Main.m

```diff
@@ -5,17 +5,34 @@
 //  Created by Marcos Magueta on 15/11/23.
 //
 
+#import "Headers/Utilities.h"
 #import "Headers/VM+Instructions.h"
 #import <Foundation/Foundation.h>
 
 int main(int argc, const char *argv[]) {
-  if (argc < 3) {
-    NSLog(@"Program and/or Keyword set is missing");
-    return 1;
-  }
   @autoreleasepool {
     VM *vm = [VM new];
-    [vm Execute:@(argv[1]) usingKeywords:@(argv[2])];
+    NSMapTable *cmds = parseCommandLine(argc, argv);
+
+    NSMutableString *candidateProgram = [[NSMutableString alloc] init];
+    NSMutableString *candidateDialect = [[NSMutableString alloc] init];
+    handleProgramAndDialect(cmds, candidateProgram, candidateDialect);
```

Comment:
This makes it mandatory to have a program in assembly, even if we provide a compiled program.